libmpv/stream_cb.h

```diff
@@ -125,6 +125,20 @@ typedef int64_t (*mpv_stream_cb_read_fn)(void *cookie, char *buf, uint64_t nbyte
  */
 typedef int64_t (*mpv_stream_cb_seek_fn)(void *cookie, int64_t offset);
 
+/**
+ * Seek by time callback used to implement a custom stream.
+ *
+ * This callback can be NULL, in which it behaves as if always returning
+ * MPV_ERROR_UNSUPPORTED.
+ *
+ * @param cookie opaque cookie identifying the stream,
+ *               returned from mpv_stream_cb_open_fn
+ * @param timestamp target timestamp position in original timebase
+ * @return the resulting offset of the stream
+ *         MPV_ERROR_UNSUPPORTED or MPV_ERROR_GENERIC if the seek failed
+ */
+typedef int64_t (*mpv_stream_cb_seek_time_fn)(void *cookie, int64_t timestamp);
```

Comment:
I'm against this, since it just exposes a libavformat implementation detail to the libmpv user. For example, what is the timebase at all? My recent experiences with AVIOContext.read_seek (which eventually results into this callback) also were rather negative: it's pretty much random whether a seek results in the callback, and it depends on the format.